OpenRA.Game/Graphics/Animation.cs

```diff
@@ -50,18 +51,21 @@ public Animation(World world, string name, Func<WAngle> facingFunc, Func<bool> p
 		}
 
 		public int CurrentFrame => backwards ? CurrentSequence.Length - frame - 1 : frame;
-		public Sprite Image => CurrentSequence.GetSprite(CurrentFrame, facingFunc());
+
+		public Sprite Image => CurrentSequence.GetSprite(CurrentFrame, facingFunc(), out rotation);
```

Comment:
The problem with introducing a separate call is that the rotation is very tightly coupled to the frame and facing parameters used in the GetSprite() call. Both of these parameters need to be used to calculate the rotation, meaning that moving the rotation out of GetSprite() would make the design much more complicated. Also from what I can see the Image function is not only used by the Render function, but also by other classes. Therefore moving the rotation to the Render() function would prevent future use of the rotation property by other classes that do not call Animation's Render() function. This current design was also suggested by pchote, and I have found it quite intuitive, so I would be reluctant to move away from it unless there is a strong reason to do so.

>Plus, for most callers to Image determining the rotation is a calculation we then throw away.

Actually the calculation only occurs for sprites with a rotation due to the 'if (InterpolatedFacings != -1)' check, so for 99% of sprites the rotation will never be calculated.

Reply:
The `rotation` is currently only used within the `Render` method, so the design that I am imaging is something roughly like:
```c#
public Sprite Image => GetImageAndRotation().Image;

(Sprite Image, WAngle Rotation) GetImageAndRotation()
{
    var frame = Frame;
    var facing = facingFunc();
    return (CurrentSequence.GetSprite(frame, facing), CurrentSequence.GetRotation(frame, facing));
}

public IRenderable[] Render(...)
{
    var (image, rotation) = GetImageAndRotation();
    var imageRenderable = new SpriteRenderable(image, ..., rotation);
    ...
}

If we ever did worry about the overhead of interpolating facings for those sprites that use them, we could add a parameter to skip it e.g.  `GetImageAndRotation(bool shouldCalculateRotation)`.

The concrete benefits are:
- Don't need to store an extra field in `Animation` for rotation.
- The `Render` method no longer relies on the side effect that `Image` must be called before `rotation`. So if anybody ever swaps those around in a seemingly safe refactoring they won't break everything.
- Avoids the overhead of calculating the interpolated rotation when calling `Image` if we don't need it. (If we have an `shouldCalculateRotation` option)

Reply:
I see what you mean. Ok I have implemented the above, removing the rotation field from Animation and returning a tuple from GetSpriteWithRotation() (I used the word Sprite to be consistent with existing terminology, and because that's the class name). I also kept the original GetSprite() call for the Image function. Since GetSprite() is no longer overloaded, it can remain as is to reduce overhead, as we are now using GetSpriteWithRotation() exclusively for the SpriteRenderable creation.